sdks/how-to/how-to-authenticate-a-user.md

```diff
@@ -0,0 +1,488 @@
+
+## Init AnonymousMedTechApi
+<!-- file://code-samples/how-to/authenticate-user/index.mts snippet:Instantiate AnonymousMedTech API-->
+```typescript
+const iCureUrl = process.env.ICURE_URL
+const msgGtwUrl = process.env.ICURE_MSG_GTW_URL
+const specId = process.env.SPEC_ID
+const authProcessByEmailId = process.env.AUTH_BY_EMAIL_PROCESS_ID
+const authProcessBySmsId = process.env.AUTH_BY_SMS_PROCESS_ID
+const recaptcha = process.env.RECAPTCHA_ID
+
+const anonymousApi = await new AnonymousMedTechApiBuilder()
+  .withICureUrlPath(iCureUrl)
+  .withCrypto(webcrypto as any)
+  .withMsgGtwUrl(msgGtwUrl)
+  .withMsgGtwSpecId(specId)
+  .withAuthProcessByEmailId(authProcessByEmailId)
+  .withAuthProcessBySmsId(authProcessBySmsId)
+  .build()
+```
+
+The [AnonymousMedTechApi](../references/classes/AnonymousMedTechApi.md) asks you to provide multiple information. Here 
+are their details :
+
+| Argument             | Description                                                                                      |
+|----------------------|--------------------------------------------------------------------------------------------------|
+| iCureUrlPath         | The URL to contact the iCure API. By default, https://api.icure.cloud is used                    |
+| msgGtwUrl            | The URL to contact the iCure Message Gateway API. By default, https://msg-gw.icure.cloud is used |
+| msgGtwSpecId         | Your iCure Message Gateway Identifier. See next section to know more about it                    |
+| authProcessByEmailId | Identifier of the authentication by email process. See next section to know more about it        |
+| authProcessBySmsId   | Identifier of the authentication by SMS process. See next section to know more about it          |
+
+As Daenaerys needs to be identified as a patient, you will have to provide the `patientAuthProcessByEmailId` as a 
+authProcessByEmailId and `patientAuthProcessBySmsId` as a authProcessBySmsId. 
+
+:::info
+
+If Daenaerys was a doctor, you would instead provide the `hcpAuthProcessByEmailId` as a
+authProcessByEmailId and `hcpAuthProcessByEmailId` as a authProcessBySmsId.
+
+:::
+
+
+## Starting the authentication process
+In iCure, you can authenticate either by email, either by SMS.
+Therefore, you will have to ask Daenaerys to provide at least an email or a mobile phone when authenticating.
+
+You will also have to implement the ReCAPTCHA mechanism and provide us the corresponding to found key during the startAuthentication
+process.
+
+:::info
+
+Check the official [reCAPTCHA v3 documentation](https://developers.google.com/recaptcha/docs/v3) for more information.
+Also, do not forget to contact the iCure team to get our ReCAPTCHA SiteKey that you will need to implement the reCAPTCHA
+
+:::
+
+
+<!-- file://code-samples/how-to/authenticate-user/index.mts snippet:Start Authentication Process By Email-->
+```typescript
+const authProcess = await anonymousApi.authenticationApi.startAuthentication(
```

Comment:
I prefer letting the `startAuthentication` / `completeAuthentication` methods as is. 
From developer point of view, the registration / login process is the same : Ensuring the user is who he meant to be. 
That way, developers can call the service without splitting the behaviour between a registration or a login if they don't want : We take care of it (Just like Authentication through Google / Facebook, ... on mobile apps). 
However, I greatly simplified the use of startAuthentication / completeAuthentication so that developers don't have to provide non useful information for the login process anymore